controllers/servers/host_definer/watcher/csi_node_watcher.py

```diff
@@ -0,0 +1,106 @@
+import time
+import os
+from threading import Thread
+from kubernetes.client.rest import ApiException
+
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers.host_definer.watcher.watcher_helper import WatcherHelper, NODES, SECRET_IDS
+from controllers.servers.host_definer.common import settings
+
+logger = get_stdout_logger()
+
+
+class CsiNodeWatcher(WatcherHelper):
+
+    def add_initial_nodes(self):
+        csi_nodes = self.csi_nodes_api.get().items
+        for csi_node in csi_nodes:
+            node_name = csi_node.metadata.name
+            if self._is_this_new_ibm_block_csi_node(csi_node) and \
+                    self.is_host_can_be_defined(node_name):
+                self.add_node_to_nodes(node_name, csi_node)
```

Comment:
so `_is_this_new_ibm_block_csi_node` only gets `csi_node`, but `add_node_to_nodes` gets both `csi_node` **and** `node_name`?
consider wrapping `self.csi_nodes_api.get().items` with a list of custom "csi_nodes" that fits the needs of this class (e.g. `csi_node.name`, `csi_node.is_new()`, etc...)

Reply:
as for wrapping the `.items` - same for other api objects (e.g. [pods](https://github.com/IBM/ibm-block-csi-driver/blob/develop/controllers/servers/host_definer/watcher/csi_node_watcher.py#L76))